controllers/storagepolicyquota/storagepolicyquota_controller.go

```diff
@@ -135,41 +135,59 @@ func (r *Reconciler) ReconcileNormal(
 		return err
 	}
 
-	// Create the StoragePolicyUsage resources.
+	// Create the StoragePolicyUsage resources for below kinds:
+	// - VirtualMachine
+	// - VirtualMachineSnapshot
+	resourceKinds := []struct {
+		Kind     string
+		NameFunc func(string) string
+	}{
+		{
+			Kind:     "VirtualMachine",
+			NameFunc: spqutil.StoragePolicyUsageNameForVM,
+		},
+		{
+			Kind:     "VirtualMachineSnapshot",
+			NameFunc: spqutil.StoragePolicyUsageNameForVMSnapshot,
+		},
+	}
+
 	var errs []error
 	for i := range objs {
-		dst := spqv1.StoragePolicyUsage{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: src.Namespace,
-				Name:      spqutil.StoragePolicyUsageName(objs[i].Name),
-			},
-		}
+		for _, resourceKind := range resourceKinds {
+			dst := spqv1.StoragePolicyUsage{
```

Comment:
I know this is not technically _your_ change, but `src` and `dst` makes it read like both of these are the same objects.  When in reality they are SPQ, and SPU.  Consider changing the name?